tests/test_bookmark.py

```diff
@@ -0,0 +1,163 @@
+    """Test tap sets a bookmark and respects it for the next sync of a stream"""
+
+    def name(self):
+        return "tap_tester_dixa_bookmark_test"
+
+    def test_run(self):
+        """
+        Verify that for each stream you can do a sync which records bookmarks.
+        That the bookmark is the maximum value sent to the target for the replication key.
+        That a second sync respects the bookmark
+            All data of the second sync is >= the bookmark from the first sync
+            The number of records in the 2nd sync is less then the first (This assumes that
+                new data added to the stream is done at a rate slow enough that you haven't
+                doubled the amount of data from the start date to the first sync between
+                the first sync and second sync run in this test)
+        Verify that for full table stream, all data replicated in sync 1 is replicated again in sync 2.
+        PREREQUISITE
+        For EACH stream that is incrementally replicated there are multiple rows of data with
+            different values for the replication key
+        """
+        
+        
+        expected_streams = self.expected_streams()
+        expected_replication_keys = self.expected_replication_keys()
+        expected_replication_methods = self.expected_replication_method()
+
+        ##########################################################################
+        # First Sync
+        ##########################################################################
+        conn_id = connections.ensure_connection(self)
+
+        # Run in check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # table and field selection
+        catalog_entries = [catalog for catalog in found_catalogs
+                           if catalog.get('tap_stream_id') in expected_streams]
+
+        self.perform_and_verify_table_and_field_selection(
+            conn_id, catalog_entries)
+
+        # Run a first sync job using orchestrator
+        first_sync_record_count = self.run_and_verify_sync(conn_id)
+        first_sync_records = runner.get_records_from_target_output()
+        first_sync_bookmarks = menagerie.get_state(conn_id)
+
+        ##########################################################################
+        # Update State Between Syncs
+        ##########################################################################
+        
+        LOGGER.info("Current Bookmark: {}".format(first_sync_bookmarks))
+        new_states = menagerie.get_state(conn_id)
+        new_states = {'currently_syncing': None,
+                      'bookmarks': {'conversations': {list(expected_replication_keys['conversations'])[0]: 1655856000000},
+                                    'messages': {list(expected_replication_keys['messages'])[0]: 1655856000000},
+                                    'activity_logs': {list(expected_replication_keys['activity_logs'])[0]: "2022-06-22T00:00:00Z"}}}
+
```

Comment:
I don't think this is a functional issue, but it's weird that we have mixed formats for the bookmark values in state.

Reply:
Earlier for consistency in bookmarking we were converting epoch timestamp into datestring format and assign it to pseudo key in each record. But after discussing it with the @cosimon, we decided to use epoch time as is for bookmarking since it is more in-line with Dixa API response and filter parameters.

Reply:
So should the activity_logs also be epoch time?

Reply:
This is Dixa API limitation, it accepts only `date_string' for `activity_logs` in filter parameters.